processor/src/trace/decoder/tests.rs

```diff
@@ -1,10 +1,574 @@
+        BlockStackTableRow::new_test(a_17, ONE, false).to_value(&alphas),
+    ];
+
+    // when JOIN operation is executed, entry for the JOIN block is added to the table
+    let mut expected_value = row_values[0];
+    assert_eq!(expected_value, p1[1]);
+
+    // when the first SPAN is executed, its entry is added to the table
+    expected_value *= row_values[1];
+    assert_eq!(expected_value, p1[2]);
+
+    // when the user op is executed, the table is not affected
+    assert_eq!(expected_value, p1[3]);
+
+    // when the first SPAN block ends, its entry is removed from the table
+    expected_value *= row_values[1].inv();
+    assert_eq!(expected_value, p1[4]);
+
+    // when the second SPAN is executed, its entry is added to the table
+    expected_value *= row_values[2];
+    assert_eq!(expected_value, p1[5]);
+
+    // when the user op is executed, the table is not affected
+    assert_eq!(expected_value, p1[6]);
+
+    // when the second SPAN block ends, its entry is removed from the table
+    expected_value *= row_values[2].inv();
+    assert_eq!(expected_value, p1[7]);
+
+    // when the JOIN block ends, its entry is removed from the table
+    expected_value *= row_values[0].inv();
+    assert_eq!(expected_value, p1[8]);
+
+    // at this point the table should be empty, and thus, all subsequent values must be ONE
+    assert_eq!(expected_value, ONE);
+    for i in 9..(p1.len() - NUM_RAND_ROWS) {
+        assert_eq!(ONE, p1[i]);
+    }
+}
+
+#[test]
+#[allow(clippy::needless_range_loop)]
+fn decoder_p1_split() {
+    let span1 = CodeBlock::new_span(vec![Operation::Mul]);
+    let span2 = CodeBlock::new_span(vec![Operation::Add]);
+    let program = CodeBlock::new_split(span1.clone(), span2.clone());
+
+    let mut trace = build_trace_from_block(&program, &[1]);
+    let alphas = rand_array::<Felt, AUX_TRACE_RAND_ELEMENTS>();
+    let aux_columns = trace.build_aux_segment(&[], &alphas).unwrap();
+    let p1 = aux_columns.get_column(P1_COL_IDX);
+
+    let a_9 = Felt::new(9);
+    let row_values = [
+        BlockStackTableRow::new_test(ONE, ZERO, false).to_value(&alphas),
+        BlockStackTableRow::new_test(a_9, ONE, false).to_value(&alphas),
+    ];
+
+    // when SPLIT operation is executed, entry for the SPLIT block is added to the table
+    let mut expected_value = row_values[0];
+    assert_eq!(expected_value, p1[1]);
+
+    // when the true branch SPAN is executed, its entry is added to the table
+    expected_value *= row_values[1];
+    assert_eq!(expected_value, p1[2]);
+
+    // when the user op is executed, the table is not affected
+    assert_eq!(expected_value, p1[3]);
+
+    // when the SPAN block ends, its entry is removed from the table
+    expected_value *= row_values[1].inv();
+    assert_eq!(expected_value, p1[4]);
+
+    // when the SPLIT block ends, its entry is removed from the table
+    expected_value *= row_values[0].inv();
+    assert_eq!(expected_value, p1[8]);
```

Comment:
Is there a reason `p1[5..8]` were skipped? The other tests have been consistent about checking all values, so I wasn't sure if this was an oversight or not.